service/matching/matching_engine.go

```diff
@@ -617,9 +617,19 @@ pollLoop:
 		if err != nil {
 			switch err.(type) {
 			case *serviceerror.Internal, *serviceerror.DataLoss:
```

Comment:
I thought about splitting this case since we *could* technically still drop Dataloss errors as CDS only seems to be emitting "Internal" in-place of "Unavailable". However, chose the more conservative approach for now.